skops/hub_utils/_hf_hub.py

```diff
@@ -643,10 +643,18 @@ def get_model_output(repo_id: str, data: Any, token: Optional[str] = None) -> An
         inputs = {"data": data.to_dict(orient="list")}
     except AttributeError:
         # the input is not a pandas DataFrame
-        inputs = {
-            col: (data[:, idx] if len(data.shape) > 1 else data[idx])
-            for idx, col in enumerate(_get_column_names(data))
-        }
+        if data.dtype.names:
+            inputs = {
+                col: (
+                    [(x[0] if len(data.shape) > 1 else x) for x in data[idx].tolist()]
```

Comment:
For converted the structured array data to a list with iteration, check the data shape for index operation.